### ISSUE TITLE
Feature Request: support for lazy_import callables

### ISSUE THREAD

**Issue by thiswillbeyourgithub** [beartype/beartype#422]
Hi,

I hope you're well!

I've been playing around with lazy loading recently, especially [the python lib lazy_import](https://github.com/mnmelo/lazy_import/).

I keep running into issues when the types of lazy loaded classes are used in a beartype decorated function.

Here's an example:
``` python
from beartype import beartype
import lazy_import

Document = lazy_import.lazy_class('langchain.docstore.document.Document')

@beartype
def my_func(adoc: Document) -> str:
    return "hi"

```

The error is:
```
beartype.roar.BeartypeDecorHintNonpepException: Function __main__.my_func() parameter "adoc" type hint <lazy_import.LazyCallable object at 0x72910ae76d90> either PEP-noncompliant or PEP-compliant but currently unsupported by @beartype. You suddenly feel encouraged to submit a feature request for this hint to our friendly issue tracker at:
	https://github.com/beartype/beartype/issues
```

So, given that the lazy_import lib has not been updated for apparently 7 years, I'm guessing maybe it's not used that much anymore so if it's not particularly easy maybe don't add support for it. But if by any chance you have any pointers as to how I could modify the function on the fly in a beartype hook to make beartype work nonetheless without triggering the issue?

When using lazy_import, when it has finally been loaded the loaded function is at the `callable` attribute btw

**Comment by leycec** [comment]
**heh.** Crazy ol' obsolete unmaintained package `lazy_import` makes a surprise comeback, huh?

We'd all *love* to see somebody pick that package back up with a friendly fork. Sadly, here we are. It is quite dead. Personally, I'd be fairly concerned with using a package *that* dead in production or even personal code.

Thankfully... @beartype fully supports what you want to do *without* requiring sketchy third-party zombie projects like `lazy_import`. Just replace your use of the lazily imported `Document` class with **PEP 484-compliant stringified forward references**: e.g.,

```python
from beartype import beartype
import lazy_import

_DOCUMENT_CLASSNAME = 'langchain.docstore.document.Document'
Document = lazy_import.lazy_class()

@beartype
def my_func(adoc: _DOCUMENT_CLASSNAME) -> str:
    return "hi"
```

**Totally works.** Rejoice, QA brethren! Alternately, if you decide you don't even need or want `lazy_import` anymore, just cut out the middleman and directly tell @beartype "what you want, what you *really really* want":

```python
from beartype import beartype

@beartype
def my_func(adoc: 'langchain.docstore.document.Document') -> str:
    return "hi"
```

**Totally works.** Rejoicing intensifies!

![](https://media1.tenor.com/m/X15e67QrANUAAAAC/the-office.gif)
<sup>*@leycec and @thiswillbeyourgithub rejoice together*</sup>

**Comment by thiswillbeyourgithub** [comment]
Oh my god this is incredible. Thanks a lot and you made me laugh. I am rejoicing!

Btw I don't remember seeing this feature when perusing the documentation, I think it might be either not mentionned or should be a bit more obvious because it sounds like a real life saver in complex weird situations like I was in.

Thanks a lot!

**Comment by leycec** [comment]
> I think it might be either not mentionned or should be a bit more obvious...

You're totally right! I actually thought the [@beartype FAQ](https://beartype.readthedocs.io/en/latest/faq) discussed forward references. Now, I am shook. The FAQ doesn't go there with a ten foot pole – *but it absolutely should.*

Let's reopen this as a friendly reminder to myself to document forward references more openly, honestly, and obviously. :sweat_smile:
